roles/servicetelemetry/tasks/component_snmp_traps.yml

```diff
@@ -0,0 +1,28 @@
+- name: Lookup template - Deployment
+  debug:
+    msg: "{{ lookup('template', './manifest_snmp_traps_deploy.j2') | from_yaml }}"
+
+- name: Set default snmp traps manifest
+  set_fact:
+    snmp_traps_manifest_deploy: "{{ lookup('template', './manifest_snmp_traps_deploy.j2') | from_yaml }}"
+  when: snmp_traps_manifest is not defined
```

Comment:
The point of the `snmp_traps_manifest` not being defined is to allow a manifest override. Since you never use the `snmp_traps_manifest` parameter you won't be able to override the template.

Reply:
I've had this in an earlier patch, it fell flat on its nose, since the template contained both a deployment and a service. This is why it's been split into two pieces.

Reply:
How did it fail? As it stands now this pattern won't work.

Reply:
it failed in the template lookup step, I can only assume the filter from_yaml failed.

Reply:
OK, well I'd move this back to the working pattern and let's see what fails. If it was an issue with `from_yaml` I would expect this to continue failing.

Reply:
After changing it back to the failure gets back.
```
ASK [servicetelemetry : Create SNMP traps instance] ***************************
task path: /opt/ansible/roles/servicetelemetry/tasks/main.yml:53

included: /opt/ansible/roles/servicetelemetry/tasks/component_snmp_traps.yml for localhost

ASK [servicetelemetry : Lookup template] **************************************
task path: /opt/ansible/roles/servicetelemetry/tasks/component_snmp_traps.yml:1

An exception occurred during task execution. To see the full traceback, use -vvv. The error was:     ^
fatal: [localhost]: FAILED! => {"msg": "Unexpected failure during module execution."}


LAY RECAP *********************************************************************
localhost                  : ok=63   changed=0    unreachable=0    failed=1    skipped=3    rescued=0    ignored=0   
```

Reply:
You still want to keep the templates in separate files. The deployment should be in `manifest_snmp_traps_deployment.j2` file, and a separate one for Service (although you can just use the `definition` parameter of the `k8s` module and write out the manifest directly inline since you're not really doing any fancy template stuff that would result in the need to have a separate file).

Also your syntax is wrong I believe on the multi-manifest setup as it should be `---` not `----` as the separator between the manifests.

You don't need a `when` statement around the `Service` part at all either. You only need the `when` for the `Deployment` manifest since that's the component that you're allowing to be overridden.

The logic is _if the CR.snmpTrapsManifest is not defined, then set snmpTrapsManifest to contents of lookup(template)_.

You don't need to wrap the `Service` definition in anything. Just make sure it is specified and added as part of `snmpTrapsEnabled: true` which is what does the `include_tasks` in the top level playbook.